src/views/common/Animation/AnimationControls/useWheelHandler.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback, useMemo, useRef } from 'react'
+import { DebounceThrottleResolver, DebounceThrottleUpdater, useThrottler } from 'util/rateLimiters'
+import { AnimationStateDispatcher } from '../AnimationStateReducer'
+
+type WheelStateProperties = {
+    wheelCount: number
+    useFineWheel: boolean
+}
+
+type WheelStateRefs = {
+    wheelCount: React.MutableRefObject<number>
+    useFineWheel: React.MutableRefObject<boolean>
+}
+
+type WheelResolverProps = {
+    dispatch: AnimationStateDispatcher<any>
+}
+
+const wheelUpdate: DebounceThrottleUpdater<WheelStateProperties, WheelStateRefs> = (refs, state) => {
+    const unchanged = (state.wheelCount === 0 &&
+                       refs.useFineWheel.current === state.useFineWheel)
+    if (!unchanged) {
+        refs.wheelCount.current += state.wheelCount
+        refs.useFineWheel.current = state.useFineWheel
+    }
+    return !unchanged
+}
+
+const wheelResolver: DebounceThrottleResolver<WheelStateRefs, WheelResolverProps> = (refs, props) => {
+    const { dispatch } = props
+    if (refs.wheelCount.current !== 0) {
+        dispatch({
+            type: 'SKIP',
+            backward: refs.wheelCount.current < 0,
+            fineSteps: Math.abs(refs.wheelCount.current),
+            frameByFrame: refs.useFineWheel.current
+        })
+    }
+    refs.useFineWheel.current = false
+    refs.wheelCount.current = 0
+}
+
+export const useThrottledWheel = (dispatch: AnimationStateDispatcher<any>) => {
+    const wheelCount = useRef<number>(0)
+    const useFineWheel = useRef<boolean>(false)
+    const refs = useMemo(() => {return { wheelCount, useFineWheel }}, [wheelCount, useFineWheel])
+    const resolverProps = useMemo(() => { return { dispatch }}, [dispatch])
+    const wheelHandler = useThrottler(wheelUpdate, wheelResolver, refs, resolverProps)
```

Comment:
The general pattern is that the throttler maintains its own internal state and handles calling of the resolver: it returns to the consumer an interface which is just a function that takes only the values that might need to be updated. The refs that store them and the resolver that persists them to the actual state have both been away.